chapters/synchronization.txt

```diff
@@ -70,27 +73,71 @@ If an operation happens-before another operation, then the first operation
 must: complete before the second operation is initiated.
 More precisely:
 
-  * Let *A* and *B* be separate sets of operations.
   * Let *S* be a synchronization command.
-  * Let *A~S~* and *B~S~* be the synchronization scopes of *S*.
-  * Let *A'* be the intersection of sets *A* and *A~S~*.
-  * Let *B'* be the intersection of sets *B* and *B~S~*.
-  * Submitting *A*, *S* and *B* for execution, in that order, will result in
-    execution dependency *E* between *A'* and *B'*.
+  * Let *A~S~* be the first synchronization scope of *S*.
+  * Let *B~S~* be the second synchronization scope of *S*.
+  * Let *A'* be the set of operations satisfying *A~S~*.
+  * Let *B'* be the set of operations satisfying *B~S~*.
+  * Submitting *S* for execution will result in an execution dependency
+    *E* between *A'* and *B'*.
```

Comment:
I think the use of predicate logic here is great. This seems intuitive and also more precise than the current spec.

I'm wondering what you think about this potential adjustment, which uses some more of the current wording:
```
* Let *A* and *B* be separate sets of operations.
* Let *S* be a synchronization command.
* Let *A~S~* be the first synchronization scope of *S*.
* Let *B~S~* be the second synchronization scope of *S*.
* Let *A'* be the set of operations in *A* satisfying *A~S~*.
* Let *B'* be the set of operations in *B* satisfying *B~S~*.
* Submitting *A*, *S* and *B* for execution, in that order, will result in
  execution dependency *E* between *A'* and *B'*.
```

The reasoning being that, at least to me, this makes it a bit more clear that `*B'*` can include operations that are submitted later. The note you have below clarifies this, which is great, but I think it would be nice if we could have this be really clear just from the main text of the spec, so that the note is less essential.

Arguably the update I'm suggesting is a bit redundant, because the predicates already "select" the operations, but I do think it reduces the need for the extra note. (My understanding is notes are meant to be more of an optional extra, and not essential for understanding the spec).

Reply:
Sorry for late answer. Yea it is somewhat annoying **B** is a temporal set. One has to go outside spacetime to see the full extent of **B**.

Notes\comments are always desperate measures. Make no mistake, I would do everyting humanely possible to try to eliminate them.

What you suggest is there originally, and what I wanted to avoid, because it makes several inaccurate preassumptions. For example, let's say **S** is a `vkWaitForFences`. What does it even mean submitting `vkWaitForFences`? There's not even any queue to submit to...